parquet/src/test/java/org/apache/iceberg/parquet/TestVariantReaders.java

```diff
@@ -879,6 +891,270 @@ public void testMixedRecords() throws IOException {
     VariantTestUtil.assertEqual(expectedThree, actualThreeVariant.value());
   }
 
+  @Test
+  public void testSimpleArray() throws IOException {
```

Comment:
I think we also need a test case where the `value` column is not in the schema to verify that the reader can handle fully shredded structures.